### PR TITLE
7738 add analytics to Firestore

### LINKED ISSUE #7738: [Sellers' Analytics] Add analytics events to Firestore

### PR DESCRIPTION
**Pull request by RemcoSimonides**
Fix #7738

### CODE REVIEW

File: libs/firebase-utils/src/lib/firestoreMigrations/0072.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      END as createdFrom,
+      users.orgId
+    FROM (
+      SELECT 
+        event_timestamp as createdAt,
+        MAX(event_name) as name,
+        MAX(if(params.key = "movieId", params.value.string_value, NULL)) as titleId,
+        MAX(if(params.key = "uid", params.value.string_value, NULL)) as userId,
+        MAX(if(params.key = "page_location", params.value.string_value, NULL)) as page_location
+      FROM \`blockframes.analytics_193045559.events_*\`,
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/firestoreMigrations/0072.ts:31]: every project will have prod data imported. Is this what you want ?

File: libs/firebase-utils/src/lib/firestoreMigrations/0072.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    FROM (
+      SELECT 
+        event_timestamp as createdAt,
+        DATE(TIMESTAMP_MICROS(event_timestamp)) as day,
+        MAX(event_name) as name,
+        MAX(if(params.key = "uid", params.value.string_value, NULL)) as userId,
+        MAX(if(params.key = "page_location", params.value.string_value, NULL)) as page_location,
+        MAX(if(params.key = "page_title", params.value.string_value, NULL)) as page_title,
+        MAX(if(params.key = "page_path", params.value.string_value, NULL)) as page_path
+      FROM \`blockframes.analytics_193045559.events_*\`,
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/firestoreMigrations/0072.ts:71]: same

File: libs/firebase-utils/src/lib/firestoreMigrations/0072.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  const [[pageViews], [otherEvents], titles] = await Promise.all([
+    executeQuery(page_view_query),
+    executeQuery(events_query),
+    getCollection<MovieDocument>(`movies`)
+  ])
+  const rows = pageViews.concat(otherEvents);
+
+  const events = rows.map(row => {
+    const title = titles.find(title => title.id === row.titleId)
+    if (!title) return
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/firestoreMigrations/0072.ts:124]: missing `;`

File: libs/firebase-utils/src/lib/firestoreMigrations/0072.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+    const event: Analytics = {
+      id: undefined, // id is added later
+      name: row.name,
+      type: 'title',
+      _meta: createDocumentMeta({
+        createdAt: new Date(+row.createdAt / 1000),
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/firestoreMigrations/0072.ts:131]: why do you need a `+` here ?

File: libs/firebase-utils/src/lib/firestoreMigrations/0072.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        ownerOrgIds: title.orgIds
+      })
+    };
+    return event;
+
+  }).filter(row => !!row)
+
+  let i = 1;
+  let batch = db.batch();
+  while(i <= events.length) {
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/firestoreMigrations/0072.ts:147]: is it possible to use `runChunks()` ?

**Review by phpgeekfr — APPROVED** [review]

File: libs/firebase-utils/src/lib/firestoreMigrations/0072.ts (hunk 4 above)
Comment (RemcoSimonides) [reply libs/firebase-utils/src/lib/firestoreMigrations/0072.ts:131]: I thought `row.createdAt` is a string but checked it again and it's an integer already

**Pull request merged**